src/Visualization/Python/PlotControlSystem.py

```diff
@@ -0,0 +1,226 @@
+            return keys
+
+    # We are only plotting the most relevant columns for now. More can be added
+    # later if it's useful for debugging
+    def extract_relevant_columns(
+        df: pd.DataFrame, name: str, component_name: str
+    ):
+        relevant_columns = ["ControlError", "DampingTimescale"]
+
+        return df[relevant_columns].add_prefix(name + component_name)
+
+    # Get a list of all control systems (excluding size) that we have from the
+    # first reductions file and a map to all of their components
+    control_systems = check_control_system_dir(reduction_files[0])
+    control_system_components = {
+        system: get_control_system_components(reduction_files[0], system)
+        for system in control_systems
+        if "Shape" not in system or with_shape
+    }
+
+    # Open every h5file. For each h5file, turn each component of each control
+    # system into a DataFrame and concat it with the large data frame
+    data = pd.DataFrame()
+    for reduction_file in reduction_files:
+        h5file = h5py.File(reduction_file)
+
+        for system in control_system_components:
+            for component in control_system_components[system]:
+                tmp_data = check_control_system_file(
+                    reduction_file, h5file, system, component
+                )
+                tmp_data = extract_relevant_columns(tmp_data, system, component)
+                data = pd.concat([data, tmp_data])
+
+    # Restrict data to bounds
+    if x_bounds:
+        data = data[(data.index >= x_bounds[0]) & (data.index <= x_bounds[1])]
+
+    # Set up plots
+    fig, axes = plt.subplots(2, 1, sharex=True)
+
+    axes = list(axes)
+
+    # Plot all components for all systems. The name for the legend differs
+    # slightly from the name in the DataFrame, just to make it prettier for the
+    # user. Since both figures will have the same lines, we only show one legend
+    for system in control_system_components:
+        for component in control_system_components[system]:
+            df_name = f"{system}{component}"
+            legend_name = (
+                system
+                if len(control_system_components[system]) == 1
+                else f"{system} {component}"
+            )
+
+            axes[0].plot(
+                data.index,
+                np.abs(data[f"{df_name}ControlError"]),
+            )
+            axes[1].plot(
+                data.index,
+                np.abs(data[f"{df_name}DampingTimescale"]),
+                label=legend_name,
+            )
+
+    # Configure the axes and legend
+    fig.set_size_inches(10, len(axes) * 4)
+    axes[0].set_ylabel("Control Error")
+    axes[1].set_ylabel("Damping Time")
+    for ax in axes:
+        ax.grid()
+        ax.set_yscale("log")
+    if x_label:
+        # Bottom plot gets x label
+        axes[1].set_xlabel(x_label)
+    if title:
+        plt.title(title)
+    axes[1].legend(loc="center left", bbox_to_anchor=(1.01, 1.1))
```

Comment:
Opinions on how to better represent the shape components are welcome. Just did the easiest thing for now to quickly get it in.

Reply:
How about an L2 norm over all shape components

Reply:
I think it will be important to be able to distinguish which component is the one that's going bad, so I don't think lumping them together would be best.

Reply:
You could plot both the L2 norm and the highest mode

Reply:
Or one plot where the components of each control system are lumped together so you have only one line each for expansion, rotation, shape, size. Plus one plot for each control system with all the components. So you have a figure with 1+num_control_systems rows and 1 col.

Reply:
Having 1+num_control_systems plots sounds a bit crowded and also hard to compare between control systems. I do like doing the L2 over all `m` of a given `l` by default and then having an option to show all `m`.